File: states.js

var IDLE = {
  id: 0,
  master: {
    values: {
      
    },
    events: {
      
    }
  },
  tablet: {
    values: {
      
    },
    events: {
      
    }
  }
};
var STATE_IDLE = 0;
var OFF = {
  id: 1,
  master: {
    values: {
      
    },
    events: {
      
    }
  },
  tablet: {
    values: {
      
    },
    events: {
      
    }
  }
};
var STATE_OFF = 1;
var MOTIONMACHINE = {
  id: 2,
  master: {
    values: {
      
    },
    events: {
      upRamp: function upRamp() { manager.sendEvent(0, 2); },
      upStairs: function upStairs() { manager.sendEvent(1, 2); },
      moveServo: function moveServo() { manager.sendEvent(2, 2); }
    }
  },
  tablet: {
    values: {
      
    },
    events: {
      
    }
  }
};
var STATE_MOTIONMACHINE = 2;

var STATES = {
  IDLE: IDLE,
  OFF: OFF,
  MOTIONMACHINE: MOTIONMACHINE
};
var manager = new Manager([IDLE, OFF, MOTIONMACHINE]);

File: MotionMachineAMIB1/states.cpp
#include "states.h"

static const StateInfo state_infos[3] = {
  {IDLE::setup, IDLE::enter, IDLE::exit, IDLE::loop, IDLE::event},
  {OFF::setup, OFF::enter, OFF::exit, OFF::loop, OFF::event},
  {MOTIONMACHINE::setup, MOTIONMACHINE::enter, MOTIONMACHINE::exit, MOTIONMACHINE::loop, MOTIONMACHINE::event}
};

static const WireValue wire_values[0] = {
  
};

MasterManager<State, 3, 0> manager(0xa4663eee, state_infos, wire_values, 0);

namespace IDLE {


void event(uint8_t ev) {
  switch (ev) {
  
  default:
    break;
  }
}


}
namespace OFF {


void event(uint8_t ev) {
  switch (ev) {
  
  default:
    break;
  }
}


}
namespace MOTIONMACHINE {


void event(uint8_t ev) {
  switch (ev) {
  case 0:
    events::upRamp();
    break;
  case 1:
    events::upStairs();
    break;
  case 2:
    events::moveServo();
    break;
  default:
    break;
  }
}


}

